library/Faktory/Job.hs

```diff
@@ -68,16 +70,24 @@ newtype JobOptions = JobOptions [JobUpdate]
 perform
   :: (HasCallStack, ToJSON arg) => JobOptions -> Producer -> arg -> IO JobId
 perform options producer arg = do
-  job <- applyOptions options =<< newJob arg
+  let
+    namespace =
+      connectionInfoNamespace
+        $ settingsConnection
+        $ clientSettings
+        $ producerClient producer
+  job <- applyOptions namespace options =<< newJob arg
   jobJid job <$ pushJob producer job
 
-applyOptions :: JobOptions -> Job arg -> IO (Job arg)
-applyOptions (JobOptions patches) = go patches
+applyOptions :: Maybe String -> JobOptions -> Job arg -> IO (Job arg)
+applyOptions namespace (JobOptions patches) = go patches
  where
+  queuePrefix = pack $ fromMaybe "" namespace
```

Comment:
I'm honestly not sure, but would it be better to this immediately on parse and always have a non-`Maybe` namespace that is just `""` in the common case?

Reply:
Yeah I added a `Namespace` newtype on `Text` and it looks much better :+1: